src/product.js

```diff
@@ -124,7 +124,7 @@ const getApplicationPath = (product) => {
   const binContent = fs.readFileSync(product.binPath, { encoding: "UTF-8" });
 
   // Toolbox case
-  const pattern = new RegExp('open -a "(.*)" "\\$@"');
+  const pattern = new RegExp('open -(n)?a "(.*)" (--args)? "\\$@"');
   const match = pattern.exec(binContent);
   if (match && match.length === 2) {
```

Comment:
as you add 2 capturing group we need to update this part too

```suggestion
  if (match && [2, 4].includes(match.length)) {
    const matchLength = match.length;
    let appPath = match[matchLength === 4 ? 2 : 1];
```

Reply:
Done, extracted variable outside of if, to reuse it.